app/components/button/index.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo} from 'react';
+import {StyleProp, Text, TextStyle, ViewStyle} from 'react-native';
+
+import {buttonBackgroundStyle, buttonTextStyle} from '@utils/buttonStyles';
+
+import TouchableWithFeedback from '../touchable_with_feedback';
+
+type Props = {
+    theme: Theme;
+    backgroundStyle?: StyleProp<ViewStyle>;
+    textStyle?: StyleProp<TextStyle>;
+    size?: ButtonSize;
+    emphasis?: ButtonEmphasis;
+    buttonType?: ButtonType;
+    buttonState?: ButtonState;
+    onPress: () => void;
+    text: string;
+}
+
+const Button = ({
```

Comment:
This is actually something I would like to refactor in other parts of the code. Sure, I probably should use `Button` instead of `TouchableWithFeedback` (I think I was mimicking some button somewhere else). But the good thing about this button is that is more standarized for us, in the sense of it always using the `buttonBackgroundStyle` and `buttonTextStyle` functions, and just ask in the props for what is really relevant (size, emphasis, button type and state).

Will update it to use the react-native-button, but I still see value on this component.